lib/release.js

```diff
@@ -0,0 +1,427 @@
+        false);
+      this.updateNodeModuleVersion(runtime, variant, versions);
+    }
+
+    // Update any REPLACEME tags in the docs.
+    const shouldUpdateREPLACEMEs = await cli.prompt(
+      'Update REPLACEME items in docs?', true);
+    if (!shouldUpdateREPLACEMEs) return this.abort();
+    await this.updateREPLACEMEs();
+
+    this.date = await cli.promptInput(
+      'Enter release date in YYYY-MM-DD format:');
+
+    // Update Changelogs
+    const shouldUpdateChangelogs = await cli.prompt(
+      'Update changelogs?', true);
+    if (!shouldUpdateChangelogs) return this.abort();
+
+    cli.startSpinner('Updating CHANGELOG.md');
+    await this.updateMainChangelog();
+    cli.stopSpinner('Updated CHANGELOG.md');
+
+    cli.startSpinner(`Updating CHANGELOG_V${versionComponents.major}.md`);
+    await this.updateMajorChangelog();
+    cli.stopSpinner(`Updated CHANGELOG_V${versionComponents.major}.md`);
+
+    await cli.prompt('Finished editing the changelogs?', false);
+
+    // Create release commit.
+    const shouldCreateReleaseCommit = await cli.prompt(
+      'Create release commit?', true);
+    if (!shouldCreateReleaseCommit) return this.abort();
+
+    // Proceed with release only after the releaser has amended
+    // it to their liking.
+    const created = await this.createReleaseCommit();
+    if (!created) {
+      await cli.prompt(
+        'Type y when you have finished editing the release commit:',
+        false);
+    }
+
+    // Open pull request against the release branch.
+    const shouldOpenPR = await cli.prompt(
+      'Push branch and open pull request?', true);
+    if (!shouldOpenPR) return this.abort();
+    this.openPullRequest();
+
+    cli.separator();
+    cli.ok(`Release preparation for ${newVersion} complete.\n`);
+    cli.info(
+      'Please proceed to Jenkins and begin the following CI jobs:\n' +
+      '      * https://ci.nodejs.org/job/node-test-pull-request/\n' +
+      '      * https://ci.nodejs.org/job/citgm-smoker/');
+    cli.info(
+      'If this release deps/v8 changes, you\'ll also need to run:\n' +
+      '      * https://ci.nodejs.org/job/node-test-commit-v8-linux/'
+    );
+  }
+
+  async promote() {
+    // TODO(codebytere): implement.
+  }
+
+  get owner() {
+    return this.config.owner || 'nodejs';
+  }
+
+  get repo() {
+    return this.config.repo || 'node';
+  }
+
+  get upstream() {
+    return this.config.upstream;
+  }
+
+  get username() {
+    return this.config.username;
+  }
+
+  async abort() {
+    const { cli, newVersion } = this;
+
+    // TODO(codebytere): figure out what kind of cleanup we want to do here.
+
+    cli.ok(`Aborted \`git node release\` for version ${newVersion}`);
+  }
+
+  getCurrentBranch() {
+    return runSync('git', ['rev-parse', '--abbrev-ref', 'HEAD']).trim();
+  }
+
+  getLastRef() {
+    return runSync('git', ['describe', '--abbrev=0', '--tags']).trim();
+  }
+
+  getChangelog() {
+    return runSync('npx', [
```

Comment:
i'm considering just adding `branch-diff` and `changelog-maker` to deps - i'm impartial as to whether we do though if others feel more strongly.

Reply:
sgtm

Reply:
I'd be much more comfortable with including them as dependencies than running via npx. At the very least you could use a child_process with `node_modules/.bin` in the path and just call the binary.